vignettes/OrganismDbi.Rmd

```diff
@@ -0,0 +1,190 @@
+
+So in the preceding section you can see that using an 
+`r Biocpkg("OrganismDbi")`
+package behaves very similarly to how you might use a `r Biocpkg("TxDb")` or an
+`r Biocpkg("OrgDb")` package. The same methods are defined, and they behave
+similarly except that they now have access to much more data than before. But
+before you make your own `r Biocpkg("OrganismDbi")` package you need to
+understand that there are few logical limitations for what can be included in
+this kind of package.
+
+* The 1st limitation is that all the annotation resources in question must
+have implemented the four methods described in the preceding section (`columns`,
+`keys`, `keytypes` and `select`).
+
+* The 2nd limitation is that you cannot have more than one example
+of each field that can be retrieved from each type of package that is included.
+So basically, all values returned by `columns` must be unique across ALL of the
+supporting packages.
+
+* The 3rd limitation is that you cannot have more than one example of
+each object type represented. So you cannot have two org packages since that
+would introduce two `r Biocpkg("OrgDb")` objects.
+
+* And the 4th limitation is that you cannot have cycles in the graph.
+What this means is that there will be a graph that represents the relationships
+between the different object types in your package, and this graph must not
+present more than one pathway between any two nodes/objects. This limitation
+means that you can choose one foreign key relationship to connect any two
+packages in your graph.
+
+With these limitations in mind, lets set up an example. Lets show how we could
+make `r Biocpkg("Homo.sapiens")`, such that it allowed access to 
+`r Biocpkg("org.Hs.eg.db")`,
+`r Biocpkg("TxDb.Hsapiens.UCSC.hg19.knownGene")` and 
+`r Biocpkg("GO.db")`.
+
+The 1st thing that we need to do is set up a list that expresses the way that
+these different packages relate to each other. To do this, we make a list that
+contains short two element long character vectors. Each character vector
+represents one relationship between a pair of packages. The names of the vectors
+are the package names and the values are the foreign keys. Please note that the
+foreign key values in these vectors are the same strings that are returned by
+the `columns` method for the individual packages. Here is an example that shows
+how `r Biocpkg("GO.db")`, `r Biocpkg("org.Hs.eg.db")` and 
+`r Biocpkg("TxDb.Hsapiens.UCSC.hg19.knownGene")` all relate to each
+other.
+
+```{r setupColData, message = FALSE, warning = FALSE}
+gd <- list(
+  join1 = c(GO.db = "GOID", org.Hs.eg.db = "GO"),
+  join2 = c(
+    org.Hs.eg.db = "ENTREZID",
+    TxDb.Hsapiens.UCSC.hg19.knownGene = "GENEID"
+  )
+)
+```
+
+So this `data.frame` indicates both which packages are connected to each other,
+and also what these connections are using for foreign keys. Once this is
+finished, we just have to call the `r Biocpkg("makeOrganism")` package function
```

Comment:
On lines 167 and 185, replace

    `r Biocpkg("makeOrganism")` package

with

    `makeOrganismPackage`
 
This is the name of a function.